app/javascript/article-form/components/__tests__/ArticleCoverImage.test.jsx

```diff
@@ -31,6 +39,59 @@ describe('<ArticleCoverImage />', () => {
     expect(uploadInput.getAttribute('type')).toEqual('file');
   });
 
+  describe('when rendered in native iOS with imageUpload support', () => {
+    beforeEach(() => {
+      global.Runtime = {
+        isNativeIOS: jest.fn((namespace) => {
+          return namespace === 'imageUpload';
+        }),
+      };
+    });
+
+    it('does not display the file input', async () => {
+      const { queryByText } = render(
+        <ArticleCoverImage mainImage="" onMainImageUrlChange={jest.fn()} />,
+      );
+      expect(queryByText(/Upload an image/i)).not.toBeInTheDocument();
+    });
+
+    it('triggers a webkit messageHandler call when isNativeIOS', async () => {
+      global.window.webkit = {
+        messageHandlers: {
+          imageUpload: {
+            postMessage: jest.fn(),
+          },
+        },
+      };
+
+      const { queryByLabelText } = render(<ArticleCoverImage mainImage="" />);
+      const uploadButton = queryByLabelText(/Upload cover image/i);
+      uploadButton.click();
+      expect(
+        window.webkit.messageHandlers.imageUpload.postMessage,
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it('handles a native bridge message correctly', async () => {
+      const onMainImageUrlChange = jest.fn();
+      const { container } = render(
+        <ArticleCoverImage
+          mainImage=""
+          onMainImageUrlChange={onMainImageUrlChange}
+        />,
+      );
+      const nativeInput = container.querySelector(
+        '#native-cover-image-upload-message',
+      );
+
+      // Fire a change event in the hidden input with JSON payload for success
+      const fakeSuccessMessage = `{ "action": "success", "link": "/some-fake-image.jpg" }`;
+      fireEvent.change(nativeInput, { target: { value: fakeSuccessMessage } });
+
+      expect(onMainImageUrlChange).toHaveBeenCalledTimes(1);
```

Comment:
This is the new test added to check for the missing method. The test simulates what the native implementation would perform (send in a message in JSON via the hidden input field).

This will trigger said method (`handleNativeMessage `) and because the "success" action is parsed correctly then in turn the cover image should change (we check for the trigger of the mocked `onMainImageUrlChange`).

Let me know if you rather see this test do things differently.